### DISCUSSION TITLE
AI-Based video translation in Microsoft Edge version 122.0.2359.0

### DISCUSSION THREAD

**Discussion by HotCakeX** [SpyNetGirl/MSEdgeFeatures#22] [Announcements]
<img width="999" alt="image" src="https://github.com/HotCakeX/MSEdgeFeatures/assets/118815227/742e5f04-09f3-4a02-9107-49e79e9e3454">

<br>

![image](https://github.com/HotCakeX/MSEdgeFeatures/assets/118815227/e3915d37-df8a-4bc0-b061-de4f0ead6bf4)

<br>

![image](https://github.com/HotCakeX/MSEdgeFeatures/assets/118815227/c19ffaa9-fa51-4982-8b9f-82d94edeef33)

<br>

You can find this feature in the following release: https://github.com/HotCakeX/MSEdgeFeatures/releases/tag/122.0.2359.0

<br>

Related feature flags: `msTranslateVideo`, `msTranslateVideoCaption`

<br>

P.S in my experience, if you have 2 GPUs, one Intel and the other Nvidia, setting Nvidia as the GPU the Edge canary uses will help make it perform better.

**Comment by Soliver84** [comment]
it doesn't load the language why is that?

**Reply by HotCakeX** [reply]
it takes few seconds for it to download the models for local AI work to translate the video. It does put substantial amount of load on the system CPU/GPU though.

I think either the feature flags' combination isn't right or the feature isn't ready just yet to complete video translation

**Reply by Soliver84** [reply]
Did it translate anything for you?

I have a RTX 4080 and AMD Ryzen 5800X3d ;)

**Comment by Soliver84** [comment]
more with transalte:

msTranslateVideo,msTranslateVideoCaption,msTranslateVideoMemoryThreshold,msTranslateVideoCoresThreshold,msTranslateVideoConfigAudioBufferMinCapacity,msTranslateVideoConfigAudioBufferMaxCapacity,msTranslateVideoConfigEnableLogging,msTranslateVideoInferencedeviceCreateFailed,msTranslateVideoPassthroughTest,msTranslateVideoPassthroughCopy,msTranslateVideoPassthroughScaling,msDetectLanguageByBingAPI

**Comment by Soliver84** [comment]
Hier sind alle Funktionen, die "Video" enthalten, basierend auf dem bisherigen Chat und den bereitgestellten Informationen:

msTranslateVideo,msTranslateVideoCaption,msVideoSuperResolution,msVideoSuperResMsVideoSuperResCapableTrigger,msVideoSuperResRightSizeTrigger,msVideoElementImgContents,msVideoSuperResolutionUI,msEdgeGpuProcessLaunchCauses,msEdgeGpuProcessLaunchSummary,msEdgeGpuProcessHighMemory,msEdgeGpuProcessHighCpu,msEdgeGpuProcessDebugging,msEdgeGpuProcessTelemetry,msEdgeGpuProcessMemoryTracking,msEdgeGpuProcessActivity,msEdgeGpuProcessMonitor,msEdgeGpuProcessLogger,msEdgeGpuProcessPriority,msEdgeGpuProcessUsage,msEdgeCopilotAccountLogicV3,msEdgeCopilotContextEnabledOverride,msEdgeCopilotShowAsOverlayCard,msEdgeCopilotVerticalTabs,msEdgeCopilotVisibleContextCalculation,msEdgeCopilotWiderDefaultWidthEnabled,msEdgeUseGPTForAISuggestions,msEdgeCopilotResizeViaTransparentWindow,msEdgeCopilotResizingEnabled,msEdgeCopilotPermanentAppTowerEnabled,msEdgeCopilotMicaBackdropEnabled,msEdgeCopilotHistoryChatEnabled,msEdgeCopilotForegroundContextForRegKeyEnabled,msEdgeCopilotCommonPinPrefEnabled,msEdgeCopilotChataDataInFeedback,msEdgeCopilotAnimation,msEdgeCopilotAltTabWorkaroundEnabled,msTranslateVideoMemoryThreshold,msTranslateVideoCoresThreshold,msTranslateVideoConfigAudioBufferMinCapacity,msTranslateVideoConfigAudioBufferMaxCapacity,msTranslateVideoConfigEnableLogging,msTranslateVideoInferencedeviceCreateFailed,msTranslateVideoPassthroughTest,msTranslateVideoPassthroughCopy,msTranslateVideoPassthroughScaling
